spec/components/user_mentee_application/action_button_builder_component_spec.rb

```diff
@@ -3,49 +3,63 @@
 require 'rails_helper'
 
 RSpec.describe UserMenteeApplication::ActionButtonBuilderComponent, type: :component do
-  let!(:mentee_application) { create(:user_mentee_application) }
+  context 'when the current user is an admin' do
+    let(:mentee_application) { create(:user_mentee_application) }
+    let(:current_user) { create(:user, :admin) }
 
-  context 'when current mentee application has a pending state' do
-    it 'renders a Promote and Reject button' do
-      component = described_class.new(mentee_application:)
-      render_inline(component)
+    context 'when current mentee application has a pending state' do
+      it 'renders a Promote and Reject button' do
+        component = described_class.new(mentee_application:, current_user:)
+        render_inline(component)
 
-      expect(page).to have_button('Promote to next stage')
-      expect(page).to have_button('Reject')
+        expect(page).to have_button('Promote to next stage')
+        expect(page).to have_button('Reject')
+      end
     end
-  end
 
-  context 'when current mentee application has a pending stage_three status' do
-    it 'renders a Promote and Reject button' do
-      mentee_application.mentee_application_states.last.stage_three!
+    context 'when current mentee application has a pending stage_three status' do
+      it 'renders a Promote and Reject button' do
+        mentee_application.mentee_application_states.last.stage_three!
 
-      component = described_class.new(mentee_application:)
-      render_inline(component)
+        component = described_class.new(mentee_application:, current_user:)
+        render_inline(component)
 
-      expect(page).to have_button('Promote to next stage')
-      expect(page).to have_button('Reject')
+        expect(page).to have_button('Promote to next stage')
+        expect(page).to have_button('Reject')
+      end
     end
-  end
 
-  context 'when current mentee application has an accepted status' do
-    it 'renders no text' do
-      mentee_application.mentee_application_states.last.accepted!
+    context 'when current mentee application has an accepted status' do
+      it 'renders no text' do
+        mentee_application.mentee_application_states.last.accepted!
 
-      component = described_class.new(mentee_application:)
-      render_inline(component)
+        component = described_class.new(mentee_application:, current_user:)
+        render_inline(component)
 
-      expect(page).to have_text('')
```

Comment:
this passes both when there are and are not buttons, therefore it is not the correct assertion. Has been updated.